recipes/volume_extend.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Cookbook Name:: gluster
+# Recipe:: server_setup
+#
+# Copyright 2015, Andrew Repton
+# Copyright 2015, Biola University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+node['gluster']['server']['volumes'].each do |volume_name, volume_values|
+  # 1. Get the current size of the logical volume
+  # 2. Compare to the size set for the gluster volume
+  # 3. If different, run a resize action against that volume
+  # ToDO: change hardcoded VG name gluster into an attribute
+  require 'lvm'
```

Comment:
This diff should fix the above issue
```
diff --git a/metadata.rb b/metadata.rb
index 9147e89..76506be 100644
--- a/metadata.rb
+++ b/metadata.rb
@@ -7,4 +7,4 @@ long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
 version          '4.0.2'
 depends          'apt', '>= 2.0'
 depends          'yum', '>= 3.0'
-depends          'lvm'
+depends          'lvm', '>= 1.5.1'
diff --git a/recipes/repository.rb b/recipes/repository.rb
index 0a7ed75..111bd4b 100644
--- a/recipes/repository.rb
+++ b/recipes/repository.rb
@@ -47,8 +47,15 @@ when 'ubuntu'
     end
   end
 when 'redhat', 'centos'
+  # CentOS 6 and 7 have Gluster in the Storage SIG instead of a gluster hosted repo
+  repo_url = if node['platform_version'].to_i > 5
+               "http://mirror.centos.org/centos/$releasever/storage/$basearch/gluster-#{node['gluster']['version']}/"
+             else
+               "http://download.gluster.org/pub/gluster/glusterfs/#{node['gluster']['version']}/LATEST/EPEL.repo/epel-$releasever/$basearch/"
+             end
+
   yum_repository 'glusterfs' do
-    url "http://download.gluster.org/pub/gluster/glusterfs/#{node['gluster']['version']}/LATEST/EPEL.repo/epel-$releasever/$basearch/"
+    url repo_url
     gpgcheck false
     action :create
   end
diff --git a/recipes/server.rb b/recipes/server.rb
index 06c3880..6891e97 100644
--- a/recipes/server.rb
+++ b/recipes/server.rb
@@ -18,6 +18,9 @@
 # limitations under the License.
 #
 
+node.set['lvm']['di-ruby-lvm']['compile_time'] = true
+node.set['lvm']['di-ruby-lvm-attrib']['compile_time'] = true
+
 include_recipe 'gluster::server_install'
 include_recipe 'lvm'
 include_recipe 'gluster::server_setup'
diff --git a/test/integration/Vagrantfile.erb b/test/integration/Vagrantfile.erb
index 334c9bd..cb6b94c 100644
--- a/test/integration/Vagrantfile.erb
+++ b/test/integration/Vagrantfile.erb
@@ -25,6 +25,11 @@ Vagrant.configure("2") do |c|
   c.vm.network "private_network", ip: "<%= config[:ipaddress] %>"
 <% end %>
 
+  c.vm.synced_folder ".", "/vagrant", disabled: true
+<% config[:synced_folders].each do |source, destination, options| %>
+  c.vm.synced_folder "<%= source %>", "<%= destination %>", <%= options %>
+<% end %>
+
   c.vm.provider :<%= config[:provider] %> do |p|
     <% if config[:additional_disk] && config[:provider] == 'virtualbox' %>
       <% disk_path = "/tmp/#{rand(36**50).to_s(36)}_disk2.vdi" %>
```